### ISSUE TITLE
Allow compute_kind for SourceAsset

### ISSUE THREAD

**Issue by stefkauff** [dagster-io/dagster#15235]
### What's the use case?

Using `dagster.@asset(compute_kind="DASK"` you can set `compute_kind`. This does not seem to be possible when defining an asset through `SourceAsset(`. 

### Ideas of implementation

Add `compute_kind`  to `SourceAsset`.

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
Our thinking for not supporting this so far has been the following: `compute_kind` is a description of how an asset is computed, but the idea of `SourceAsset`s is kind of that we don't know how they're computed - i.e. they're computed outside of the code location where they're defined.

**Comment by stefkauff** [comment]
Currently we are using the `SourceAsset` to indicate we're fetching data from an external source, which is not capable of dealing with the typical dagster partitioning and materialization. 

The external data is being fetched by using a scheduled `@job`, in turn calling an accompanied `@op`. The output of the `@op` where the fetching and a calculation is done, is saved by the IO_MANAGER, which ties it together. The `"SourceAsset"` is then the source for further downstream processing (we are using a multi-stage setup).

Graphically it would look like this.

Happy to hear if we can improve our approach. Many thanks.

<img width="509" alt="image" src="https://github.com/dagster-io/dagster/assets/110971284/40082dd0-ff96-4556-a419-bf1079ef126a">

**Comment by sryza** [comment]
> The external data is being fetched by using a scheduled @job

I see. Why not use an asset for this?

**Comment by stefkauff** [comment]
I was told that would enable the materialize button and will "request" materialization as by partitioning, something which the external source does not support. Are we using it wrong?

**Comment by ChristopherFyfe** [comment]
One reason we took the above approach using a source asset over an asset was we wanted to prevent attempts to materialise or backfill data through the UI for partitions that might no longer be available from the external source.  But also we were trying to avoid the data being overwritten if the logic in the underlying op was changed.

We're still learning how to use dagster and thought that using a source asset was the best way to accomplish the above.  Is there a better way to handle this use case?

**Comment by sryza** [comment]
@stefkauff 

> and will "request" materialization as by partitioning, something which the external source does not support.

Are you able to explain this in a little more detail? Is the issue that, when you run the op to update this asset, you don't know ahead of time which partitions you're going to update? So it dynamically logs `AssetMaterialization`s based on the data that it ends up processing?

> I was told that would enable the materialize button

If we were to somehow make it possible for the materialize button to trigger the same computation that your current op runs, would that make this less of a problem?

**Comment by sryza** [comment]
@ChristopherFyfe - is this relevant to your issue? https://github.com/dagster-io/dagster/issues/12253

**Comment by ChristopherFyfe** [comment]
> @ChristopherFyfe - is this relevant to your issue? #12253

Yes exactly - the setup @stefkauff is describing was our attempt to work around this.  We would like to prevent attempts to backfill data for partitions where the source data is no longer available, so we chose to use source assets because they don't display the materialise button.

**Comment by ion-elgreco** [comment]
@sryza even if it's computed outside of dagster it still makes sense to add a compute kind, for example I want to add compute_kind "delta", to mark these as upstream delta tables sources.
